libs/libfoundation/include/libfoundation/AssetManager.h

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2020-2022 The opuntiaOS Project Authors.
+ *  + Contributed by Mainasara Tsowa <tsowamainasara@gmail.com>
+ *
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+
+// Begining of possible AssetManager to locate/read/write to files and folders in the app directory
+namespace LFoundation {
+
+class AssetManager {
+public:
+    AssetManager(const std::string& name)
+        :m_app_root(std::string() + "/Applications/" + name + ".app/Content") 
```

Comment:
It would be better if we could look for Content folder getting the binary path. I would to implement this similar to [Apple's Bundle](https://developer.apple.com/documentation/foundation/bundle). They have the similar to what you have implemented. `bundle.main` could be initialised with a binary path (see `ProcessInfo::parse_info_file`) during the startup of UI Apps (again as a [ProcessInfo](https://github.com/opuntiaOS-Project/opuntiaOS/blob/master/libs/libui/src/main.cpp#L15)).

Both UI and Foundation libs I try to make as close as possible to Apple's versions (as I have plans on launching recompiled iOS apps one day). So it would be great that all APIs have the similar behaviour to Apple's.